eclipse-collections-code-generator/src/main/resources/impl/stream/primitive/primitiveSingletonSpliterator.stg

```diff
@@ -0,0 +1,69 @@
+import "copyrightAndOthers.stg"
+
+forSpecializedStream() ::= "true"
+
+targetPath() ::= "org/eclipse/collections/impl/stream/primitive"
+
+fileName(primitive) ::= "<primitive.name>SingletonSpliterator"
+
+
+class(primitive) ::= <<
+<body(primitive.type, primitive.name)>
+>>
+
+body(type, name) ::= <<
+<copyrightAndOthers()>
+
+package org.eclipse.collections.impl.stream.primitive;
+
+import java.util.Spliterator;
+import java.util.function.<name>Consumer;
+
+/**
+ * <name>SingletonSpliterator is a specialized spliterator of type <type>.
+ * This file was automatically generated from template file primitiveSingletonSpliterator.stg.
+ *
+ * @since ...
+ */
+public final class <name>SingletonSpliterator implements Spliterator.Of<name>
+{
+    final <type> elem;
+    int est = 1;
```

Comment:
Please make the fields private.

The name `est` isn't obvious to me. Does it refer to estimateSize? I think the state you need is:
```java
private boolean visited = false;
```
or hasVisited, or hasAdvanced, or something like that.

Reply:
i need also to return the size which is first 1 and then 0. This is the same as in the collections singleton spliterator

Reply:
We can't copy/paste from OpenJDK into Eclipse Collections this way. They don't have the same OSS license. I realize that there's only one way to create an empty spliterator and thus the code will look identical. In this case, it's clear that they are the same.

Reply:
so what should i do? I can change the name of the variable from est to estimatedSize, is this enough? The logic is not too complex so there is not a lot of room to play. I can make a bit different implementation with visited false and then write in size function. if visited return 0 else return 1. is this better from license viewpoint? I think that from performance they are the same

Reply:
It's a tricky question. I didn't know about / didn't read the similar implementation when I wrote the comments about using a boolean instead. I honestly think it's a good implementation, not just a way to create an arbitrary difference.